pkg/controller/acmeorders/sync.go

```diff
@@ -505,24 +506,46 @@ func (c *controller) finalizeOrder(ctx context.Context, cl acmecl.Interface, o *
 		derBytes = block.Bytes
 	}
 
+	// Call to CreateOrderCert finalizes the ACME order. This call can only be made once.
 	certSlice, certURL, err := cl.CreateOrderCert(ctx, o.Status.FinalizeURL, derBytes, true)
-	// if an ACME error is returned and it's a 4xx error, mark this Order as
-	// failed and do not retry it until after applying the global backoff.
-	if acmeErr, ok := err.(*acmeapi.Error); ok {
-		if acmeErr.StatusCode >= 400 && acmeErr.StatusCode < 500 {
-			log.Error(err, "failed to finalize Order resource due to bad request, marking Order as failed")
-			c.setOrderState(&o.Status, string(cmacme.Errored))
-			o.Status.Reason = fmt.Sprintf("Failed to finalize Order: %v", err)
-			return nil
+
+	acmeErr, ok := err.(*acmeapi.Error)
+
+	// If finalizing the order returns a 4xx error then either we have
+	// already finalized this order or the order can be considered failed.
+	if ok && acmeErr.StatusCode >= 400 && acmeErr.StatusCode < 500 {
+
+		// Check if finalizing the ACME order failed because it is
+		// already valid. This scenario is possible if the ACME order
+		// has already been finalized in an earlier reconcile, but the
+		// reconciler failed to update the status of the Order CR.
+		acmeOrder, getOrderErr := getACMEOrder(ctx, cl, o)
+		if acmeGetOrderErr, ok := getOrderErr.(*acmeapi.Error); ok {
+			if acmeGetOrderErr.StatusCode >= 400 && acmeGetOrderErr.StatusCode < 500 {
+				log.Error(err, "failed to retrieve the ACME order (4xx error) marking Order as failed")
+				c.setOrderState(&o.Status, string(cmacme.Errored))
+				o.Status.Reason = fmt.Sprintf("Failed to retrieve Order resource: %v", err)
+				return nil
+			}
 		}
+		if getOrderErr != nil {
+			return getOrderErr
```

Comment:
question: when is this error going to happen, is it a transient error e.g. network issue?

Reply:
> is it a transient error e.g. network issue?

Most likely 👍🏼